### ISSUE TITLE
Windows exception when using ProactorEventLoop

### ISSUE THREAD

**Issue by espositofulvio** [aio-libs/aiomysql#71]
On Windows, using ProactorEventLoop raises an exception when trying to create a pool or a connection:

```python
import asyncio
import aiomysql
import sys


async def mysql_example( loop ):
    async with aiomysql.create_pool(host='ensembldb.ensembl.org', port=3306, user='anonymous', password='', db='danio_rerio_vega_78_9', loop=loop) as pool:
        async with pool.get() as conn:
            async with conn.cursor() as cur:
                await cur.execute("SELECT 42;")
                value = await cur.fetchone()
                print(value)



def main():
    loop = asyncio.get_event_loop()
    if sys.platform == 'win32':
        loop = asyncio.ProactorEventLoop()
        asyncio.set_event_loop( loop )

    loop.run_until_complete( mysql_example( loop ) )

if __name__ == '__main__':
    main()
```

The db is a public accessible mysql server I found on the internet just for writing a quick test.

The TraceBack is:

Exception in callback _ProactorReadPipeTransport._loop_reading(<_OverlappedF...op_reading()]>)
handle: <Handle _ProactorReadPipeTransport._loop_reading(<_OverlappedF...op_reading()]>)>
Traceback (most recent call last):
  File "D:\Programs\Anaconda\envs\siope\lib\asyncio\events.py", line 125, in _run
    self._callback(*self._args)
  File "D:\Programs\Anaconda\envs\siope\lib\asyncio\proactor_events.py", line 183, in _loop_reading
    data = fut.result()  # deliver data later in "finally" clause
  File "D:\Programs\Anaconda\envs\siope\lib\asyncio\futures.py", line 268, in result
    raise InvalidStateError('Result is not ready.')
asyncio.futures.InvalidStateError: Result is not ready.
Exception in callback _ProactorReadPipeTransport._loop_reading(<_OverlappedF...S&a;?is*\x00'>)
handle: <Handle _ProactorReadPipeTransport._loop_reading(<_OverlappedF...S&a;?is*\x00'>)>
Traceback (most recent call last):
  File "D:\Programs\Anaconda\envs\siope\lib\asyncio\events.py", line 125, in _run
    self._callback(*self._args)
  File "D:\Programs\Anaconda\envs\siope\lib\asyncio\proactor_events.py", line 181, in _loop_reading
    self._closing)
AssertionError

I'll investigate myself in the near future but maybe someone else has a hunch why this is happening?

**Comment by espositofulvio** [comment]
I believe this is an issue in ProactorEventLoop. Calling pause_reading/resume_reading when setting the no delay flag on the socket would reschedule the internal _loop_reading function a second time and as soon as a read is issued future.result() is called twice and the second time it raises an InvalidStateException.

**Comment by jettify** [comment]
Thanks for letting know! May be you can raise issue in asycncio bug tracker?

**Comment by espositofulvio** [comment]
Already done: http://bugs.python.org/issue26819

**Comment by espositofulvio** [comment]
Just to let you know, the Python Core team has a fix for this issue in the python 3.7 branch that is also been backported to 3.6. No luck for 3.5 I guess.

**Comment by asvetlov** [comment]
Sorry, no. Python 3.5 is in security mode; it means that only critical security bugs are backported into this branch.
